ndd_slurm.py

```diff
@@ -63,14 +65,31 @@ def get_slave_parser():
     return parser
 
 
-def get_master_ndd_cmd(args):
-    cmd = [args.n, '-i', args.i, '-s', '{}:{}'.format(args.s, args.p)]
+def get_master_ndd_cmd_head(args, r):
```

Comment:
Please clarify `r`, maybe is should be named `read_pipe` or something like that.